pkg/http_endpoint/v1/router.go

```diff
@@ -56,26 +60,645 @@ func authRouter(clients *common.ClientContainer,
+			common.WriteErrorToResponse(w, http.StatusUnauthorized,
+				"Unauthorized", err.Error())
+			return
+		// If any other error happened -> return 500 error
+		default:
+			log.Logger.Error(err)
+			common.WriteErrorToResponse(w, http.StatusInternalServerError,
+				http.StatusText(http.StatusInternalServerError),
+				"Internal server error occured")
+			return
+		}
+	}
+	dslist, err := clients.Grafana.GetDataSourceList()
+	if err != nil {
+		log.Logger.Error(err)
+		common.WriteErrorToResponse(w, http.StatusInternalServerError,
+			http.StatusText(http.StatusInternalServerError),
+			"Internal server error occured")
+		return
+	}
+	datasources, _ := json.Marshal(dslist)
+	w.WriteHeader(http.StatusOK)
+	w.Write(datasources)
+		}
+}
+
+// GetDatasourceID method gets the datasource by ID
+func GetDatasourceID(clients *common.ClientContainer) http.HandlerFunc {
+                return func(w http.ResponseWriter, r *http.Request) {
+	err := clients.Grafana.DoLogon()
+        if err != nil {
+                switch err.(type) {
+                // common.InvalidUsernamePassword means, that user provided invalid
+                // username/password. We have to return 401 error then
+                case common.InvalidUsernamePassword:
+                        common.WriteErrorToResponse(w, http.StatusUnauthorized,
+                                "Unauthorized", err.Error())
+                        return  
+                // If any other error happened -> return 500 error
+                default:
+                        log.Logger.Error(err)
+                        common.WriteErrorToResponse(w, http.StatusInternalServerError,
+                                http.StatusText(http.StatusInternalServerError),
+                                "Internal server error occured")
+                        return  
+                }       
+        }
+	datasourceID := chi.URLParam(r, "datasourceID")
+	ID, _ := strconv.Atoi(datasourceID)
+	dslist, err := clients.Grafana.GetDataSourceListID(ID)
+	if err != nil {
+                switch err.(type) {
+                // common.DatasourceNotFound means, that user provided an ID with
+                // datasource that doesn't exist. We have to return 404 error then
+                case common.DatasourceNotFound:
+                        common.WriteErrorToResponse(w, 404,
+                                "Datasource not exist", err.Error())
+                        return  
+                // If any other error happened -> return 500 error
+                default:
+                        log.Logger.Error(err)
+                        common.WriteErrorToResponse(w, http.StatusInternalServerError,
+                                http.StatusText(http.StatusInternalServerError),
+                                "Internal server error occured")
+                        return
+                }		
+	}
+	datasources, _ := json.Marshal(dslist)
+	w.WriteHeader(http.StatusOK)
+	w.Write(datasources)
```

Comment:
according to spec -> this is correct value to return
{
  "id": 0,
  "type": "influxdb",
  "name": "string",
  "configuration": "string",
  "secrets": "string"
}